pycausal_explorer/tlearner/_two_learner.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+from sklearn.base import clone
+from sklearn.utils.validation import check_is_fitted, check_X_y
+
+from pycausal_explorer.base import BaseCausalModel
+
+
+class TLearner(BaseCausalModel):
+    """
+    Implementation of the two learner model, also known as grouped conditional outcome model (GCOM).
+
+    Uses a provided model to predict outcome under treatment, and another model to predict outcome when
+    not under treatment.
+    Uses both models to estimate treatment effect.
+
+    Parameters
+    ----------
+    treatment_learner: estimator object
+        base learner to use when predicting outcome for samples under treatment.
+
+    control_learner: estimator object
+        base learner to use when predicting outcome for samples not under treatment.
+    """
+
+    def __init__(self, treatment_learner, control_learner):
+        if isinstance(treatment_learner, type) or isinstance(control_learner, type):
+            raise ValueError(
+                "You should provide an instance of an estimator instead of a class."
+            )
+        else:
+            self.treatment_learner = clone(treatment_learner)
+            self.control_learner = clone(control_learner)
+
+    def fit(self, X, y, *, treatment):
+        X, y = check_X_y(X, y)
+        X, w = check_X_y(X, treatment)
+
+        self.fitted_treatment_model = self.treatment_learner.fit(X[w == 1], y[w == 1])
```

Comment:
No need to use different variable for fitted models, use 
```suggestion
        self.treatment_learner = self.treatment_learner.fit(X[w == 1], y[w == 1])
```